src/wrappers/loadable.js

```diff
@@ -0,0 +1,63 @@
+import React, { useEffect } from 'react'
+import { Router, Route, useParams } from 'react-router-dom'
+import { bindActionCreators } from 'redux'
+import { connect, useSelector } from 'react-redux'
+import { restoreDepositState, restoreRedemptionState } from '../actions';
+import { useWeb3React } from '@web3-react/core'
+
+export const RESTORER = {
+  DEPOSIT: 'deposit',
+  REDEMPTION: 'redemption'
+}
+
+function LoadableBase({ children, account, restoreDepositState, restoreRedemptionState, restorer }) {
+    // Wait for web3 connected
+    const { active: web3Active } = useWeb3React()
+    const { address } = useParams()
+    const depositStateRestored = useSelector((state) => state[restorer].stateRestored)
+    
+    useEffect(() => {
+        if(web3Active) {
+            if (address && ! depositStateRestored) {
+                if (restorer == RESTORER.DEPOSIT) {
+                    restoreDepositState(address)
+                } else if (restorer == RESTORER.REDEMPTION) {
+                    restoreRedemptionState(address)
+                } else {
+                    throw "Unknown restorer."
+                }
+                // return <div>Loading...</div>
+            } else {
+                // FIXME How do we not render these if we're getting ready to transition to
+                // FIXME a new page?
```

Comment:
Did this rework happen to fix this FIXME?

Also, I think we can drop the commented-out returns in here, no?